### PR TITLE
Lis3dh dts

### PR DESCRIPTION
**Pull request by avisconti**
Add dts support to lis3dh sensor (#9962)

### CODE REVIEW

**Comment by avisconti** [comment]
Tested as external I2C sensor on ArgonKey board.

File: dts/bindings/sensor/st,lis3dh-i2c.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+description: >
+    This binding gives a base representation of LIS3DH 3-axis accelerometer
+
+inherits:
+    !include i2c-device.yaml
+
+properties:
+    compatible:
+        constraint: "st,lis3dh-i2c"
```
Comment (galak) [inline dts/bindings/sensor/st,lis3dh-i2c.yaml]: Since we only support the i2c mode right now, lets just call this "st,lis3dh"

**Review by galak — CHANGES_REQUESTED** [review]
Rename "st,lis3dh-i2c" to "st,lis3dh" as we only support the I2C mode right now

File: dts/bindings/sensor/st,lis3dh-i2c.yaml (hunk 1 above)
Comment (avisconti) [reply dts/bindings/sensor/st,lis3dh-i2c.yaml]: Actually I did it in this way because we may want to support spi as well soon.

Comment (avisconti) [reply dts/bindings/sensor/st,lis3dh-i2c.yaml]: But maybe not that soon.

Comment (avisconti) [reply dts/bindings/sensor/st,lis3dh-i2c.yaml]: ok, repushed

**Review by galak — APPROVED** [review]
Looks good, do we want to rename st,lis3dh.yaml to st,lis3dh-i2c.yaml right now or wait til SPI support is added?

**Comment by avisconti** [comment]
> Looks good, do we want to rename st,lis3dh.yaml to st,lis3dh-i2c.yaml right now or wait til 
> SPI support is added?

I can move it back to st,lis3dh-i2c.yaml but keeping the compatible string as it is now.

**Comment by galak** [comment]
> I can move it back to st,lis3dh-i2c.yaml but keeping the compatible string as it is now.

agreed.

**Comment by avisconti** [comment]
> I can move it back to st,lis3dh-i2c.yaml but keeping the compatible string as it is now.
done

**Comment by avisconti** [comment]
recheck

**Comment by avisconti** [comment]
recheck

**Comment by avisconti** [comment]
@galak  @MaureenHelm 
Any further comment?

**Review by nashif — APPROVED** [review]

**Pull request merged**